src/movepick.cpp

```diff
@@ -106,27 +106,16 @@ void MovePicker::score() {
   static_assert(Type == CAPTURES || Type == QUIETS || Type == EVASIONS, "Wrong type");
 
   for (auto& m : *this)
-      if (Type == CAPTURES)
+      if (Type == CAPTURES || (Type == EVASIONS && pos.capture(m)))
```

Comment:
Does bench change if you use this for line 109?

````
if (pos.capture(m))
````

Reply:
yes, the capture stage also generates queen promotions, but pos.capture doesn't flag them as a capture (and pos.capture_or_promotion, would be true also for underpromotions, which the capture stage doesn't generate... we have no function that matches 'as generated by the capture stage').

Reply:
OK, thanks!

Reply:
Err, does this mean that in current master we use the following code in lines 109-111 to score pawns promoting to queens even when the promotion is a non-capture ?

```
   if (Type == CAPTURES)
       m.value =  PieceValue[MG][pos.piece_on(to_sq(m))]
    + (*captureHistory)[pos.moved_piece(m)][to_sq(m)][type_of(pos.piece_on(to_sq(m)))] / 8;
````

Reply:
Yes. It might not matter so much since the capture history is being updated as:
```
      if (pos.capture_or_promotion(move))
          captureHistory[moved_piece][to_sq(move)][captured] << bonus;
```
so the promotions (but also underpromotions, which are quiets) influence the captureHistory.

Reply:
@snicolet I once had a PR that tried to make this more consistent, but it was not merged (was maybe also not complete): https://github.com/official-stockfish/Stockfish/pull/1401